### PR TITLE
mrc-447 Added schema for data filters

### PR DESCRIPTION
**Pull request by EmmaLRussell**

### CODE REVIEW

File: inst/schema/FilterOption.schema.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "properties": {
+    "name": {
+      "type": "string"
+    },
+    "options": {
+      "type": "array",
+      "items": {
+        "$ref": "#"
```
Comment (EmmaLRussell) [inline inst/schema/FilterOption.schema.json]: Options can contain other options, allowing for multiple geo levels

File: inst/schema/ValidateInputRequest.schema.json [hunk 2]
```diff
@@ -3,7 +3,8 @@
   "type": "object",
   "properties": {
     "type" : { "$ref": "InputType.schema.json" },
-    "path" : { "$ref": "FilePath.schema.json" }
+    "path" : { "$ref": "FilePath.schema.json" },
+    "filters": { "$ref": "Filters.schema.json" }
```
Comment (EmmaLRussell) [inline inst/schema/ValidateInputRequest.schema.json]: Not requiring this property, so shouldn't break current API!

File: inst/schema/NestedFilterOption.schema.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (EmmaLRussell) [inline inst/schema/NestedFilterOption.schema.json]: Most filter options can be simple string arrays, but this allows for options which contain further options

File: inst/schema/SurveyResponseData.schema.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (EmmaLRussell) [inline inst/schema/SurveyResponseData.schema.json]: Added this just as an example response type with filters in, and so I can get some test data being passed back to the front end initially

File: inst/schema/ANCFilters.schema.json [hunk 5]
```diff
@@ -0,0 +1,20 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "properties": {
+    "age": {
```
Comment (EmmaLRussell) [inline inst/schema/ANCFilters.schema.json]: ANC is all female, so no sex filter required here

File: inst/schema/SurveyFilters.schema.json [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+      }
+    },
+    "survey": {
+      "type": "array",
+      "items": { "type": "string" }
+    }
+  },
+  "additionalProperties": false,
+  "required": [
+    "sex", "age", "region"
```
Comment (EmmaLRussell) [inline inst/schema/SurveyFilters.schema.json]: Survey is optional - not returned if file contains only one survey

File: inst/schema/SurveyFilters.schema.json [hunk 7]
```diff
@@ -0,0 +1,28 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "properties": {
+    "sex": {
+      "type": "array",
+      "items": { "type": "string" }
```
Comment (hillalex) [inline inst/schema/SurveyFilters.schema.json]: Will sex not always be "female", "male", "both"?

File: inst/schema/SurveyFilters.schema.json [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    },
+    "age": {
+      "type": "array",
+      "items": { "type": "string" }
+    },
+    "region": {
+      "type": "array",
+      "items": {
+        "$ref": "NestedFilterOption.schema.json"
```
Comment (hillalex) [inline inst/schema/SurveyFilters.schema.json]: Will these differ between survey, program and ANC? Or can we just have a master list of regions for a country (if some regions are missing for a particular data set they just appear as grey/no data in the leaflet map)

File: inst/schema/SurveyFilters.schema.json (hunk 7 above)
Comment (EmmaLRussell) [reply inst/schema/SurveyFilters.schema.json]: I think we can probably enforce that, so yes, in that case, these could be hardcoded. (?)

File: inst/schema/SurveyFilters.schema.json (hunk 8 above)
Comment (EmmaLRussell) [reply inst/schema/SurveyFilters.schema.json]: Yeah, I was wondering that too, this is potentially quite a bulky data structure to pass around multiple times, just on the offchance that one might contain a slightly reduced list. We're expecting the norm to be that all files have data from all areas (?). So if we did just have one master list, maybe it should be the shape file validate that returns that?

Comment (hillalex) [reply inst/schema/SurveyFilters.schema.json]: Yeah, I feel like returning this with the shapefile would make sense. In the case of missing data regions will just show as greyed out on the map, can have a legend to indicate that means "no data".

File: inst/schema/NestedFilterOption.schema.json [hunk 9]
```diff
@@ -8,7 +8,7 @@
     "options": {
       "type": "array",
       "items": {
-        "$ref": "#"
+        "type": "object"
```
Comment (EmmaLRussell) [inline inst/schema/NestedFilterOption.schema.json]: Annoyingly the generateTypeScript task in hint falls over with circular references in the json schema, whatever option you choose for supposedly dealing with them.

**Pull request merged**